com.ibm.streamsx.topology/opt/python/packages/streamsx/scripts/sc.py

```diff
@@ -19,10 +23,88 @@ def main(args=None):
+    info = '{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}'
+    common = '{http://www.ibm.com/xmlns/prod/streams/spl/common}'
+    dependency_elements = root.find(info + 'dependencies').findall(info + 'toolkit')
+    for dependency_element in dependency_elements:
+        name = dependency_element.find(common + 'name').text
+        version = dependency_element.find(common + 'version').text
+        deps[name] = version
+    return deps
+
+def add_local_toolkits(toolkits, dependencies, topo):
+    """ A helper function that given the local toolkit paths, and the dependencies (ie toolkits that the apps depends on),
+    checks and adds the local toolkit if it is one of the dependencies
+
+    Arguments:
+        toolkits {List} -- A list of local toolkit directories, each could be a toolkit directory (has info.xml directly inside),
+                or a directory consisting of toolkits (no info.xml directly inside)
+        dependencies {Dictionary} -- A dictionary where the key is the names of the dependencies, and the value is its version.
+        topo {topology Object} -- [description]
+    """
+    local_toolkits = [] # toolkit objects
+    #  For each path, check if its a SPL toolkit (has info.xml directly inside) or a directory consisting of toolkits (no info.xml directly inside)
+    for x in toolkits:
+        if _is_toolkit_(x):
+            tk = _get_toolkit(x)
+            local_toolkits.append(tk)
+        else:
+            # directory consisting of toolkits, get list and check which are local_toolkits, add them
+            sub_directories = glob(x + "*/")
+            for y in sub_directories:
+                if _is_toolkit_(y):
+                    tk = _get_toolkit(y)
+                    local_toolkits.append(tk)
+
+    # Once we have all local toolkits, check if they correspond to any dependencies that we have, if so, add them
+    for toolkit in local_toolkits:
+        if toolkit.name in dependencies:
+            # print(toolkit.name, toolkit.path)
+            add_toolkit(topo, toolkit.path)
+
+def _get_toolkit(path):
+    # Get the name & version of the toolkit that is in the directory PATH
+    root = ET.parse(path + "/info.xml").getroot()
+    identity = root.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}identity')
+    name = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}name')
+    version = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}version')
+
+    toolkit_name = name.text
+    toolkit_version = version.text
+    return toolkit(toolkit_name, toolkit_version, path)
+
+def _is_toolkit_(tkdir):
+    # Checks if tkdir is a toolkit (contains toolkit.xml or info.xml)
+    # or if it isn't or is directory containing multiple toolkit directories (returns)
+    for fn in ['toolkit.xml', 'info.xml']:
+        if os.path.isfile(os.path.join(tkdir, fn)):
+            return True
+    return False
+
+class toolkit:
```

Comment:
Suggest `_LocalToolkit` as class name to differentiate from REST `Toolkit` class that will be needed in subsequent steps.